### PR TITLE
policy implications

### PR DESCRIPTION
**Pull request by dpa9694**

### CODE REVIEW

File: code/src/merge.py [hunk 1]
```diff
@@ -140,7 +140,11 @@ def get_intensities(policies, adm_level):
 
         total_intensity += additional_policy_intensities.sum()
 
-    assert total_intensity <= 1
+    if total_intensity > 1:
+        print(total_intensity)
+        print(policies)
+        print(adm_level)
+        assert total_intensity <= 1
```
Comment (bolliger32) [inline code/src/merge.py:147]:
```suggestion
    assert total_intensity <=1, f"{total_intensity}, {policies}, {adm_level}"
```

File: code/data/multi_country/convert-policies-raw-to-interim.py [hunk 2]
```diff
@@ -0,0 +1,150 @@
+import pandas as pd
+import numpy as np
+import src.utils as cutil
+import json
+import operator
+import argparse
+
+
+def policy_conversion_parser():
```
Comment (bolliger32) [inline code/data/multi_country/convert-policies-raw-to-interim.py:9]: curious why you put this in a dictionary. I think it works fine! Just hadn't seen this approach before so am curious :)

File: code/data/multi_country/convert-policies-raw-to-interim.py [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+
+
+def apply_rule(df, src_policy, op_str, src_val, dst_rule, country_code):
+
+    dst_policy, dst_val = dst_rule
+
+    # Get operator function from operator string
+    op = op_dict[op_str]
+
+    if print_logs:
```
Comment (bolliger32) [inline code/data/multi_country/convert-policies-raw-to-interim.py:57]: no need to do this right now but sometime feel free to lookup the logging functionality in python. It's kind of a beast but can provide some useful functionality. ...but def not something we should do right now

File: code/data/multi_country/convert-policies-raw-to-interim.py [hunk 4]
```diff
@@ -0,0 +1,150 @@
...
+    for c in intensity_cols:
+        src_group_mask = (src_group_mask) | (psrc[c] == src_group)
+
+    psrc = psrc[src_group_mask].copy()
+
+    for dst_policy in dst_policies:
+        dst_category, dst_group = dst_policy.split(".")
+        pcopy = psrc.copy()
+        pcopy["policy"] = dst_category
+        pcopy[intensity_cols[0]] = dst_group
```
Comment (bolliger32) [inline code/data/multi_country/convert-policies-raw-to-interim.py:99]: I don't think this makes a diff but do we want to set the other `intensity_cols` to nan here? Theres no reason we would be double-counting policies this way is there?

File: code/data/multi_country/convert-policies-raw-to-interim.py [hunk 5]
```diff
@@ -0,0 +1,150 @@
...
+        pcopy["policy_intensity"] = dst_val
+
+    df = pd.concat([df, pcopy], ignore_index=True).sort_values(
+        "date_start", ascending=True
+    )
+
+    return df
+
+
+def apply_usa_rule(df, src_policy, dst_policies):
```
Comment (bolliger32) [inline code/data/multi_country/convert-policies-raw-to-interim.py:81]: Are we handling trailing spaces well here? Just confirming. Also should we drop anything that has null intensity_group within this function? (note that when I called `pd.read_csv` on the usa policy csv, one row that was NA was actually converted to `np.nan` but others were not (becaues they had trailing spaces, e.t. "NA ")

File: code/data/multi_country/convert-policies-raw-to-interim.py (hunk 3 above)
Comment (dpa9694) [reply code/data/multi_country/convert-policies-raw-to-interim.py:57]: yep i know it's there, need to learn it!

File: code/data/multi_country/convert-policies-raw-to-interim.py (hunk 4 above)
Comment (dpa9694) [reply code/data/multi_country/convert-policies-raw-to-interim.py:99]: oh I think that does make a difference, good catch! I think we should assign all other `intensity_cols` to nan, will add that in

File: code/data/multi_country/convert-policies-raw-to-interim.py (hunk 2 above)
Comment (dpa9694) [reply code/data/multi_country/convert-policies-raw-to-interim.py:9]: a function? no real reason, i just thought it's a little easier on the eyes to group those things in an indent

File: code/data/multi_country/convert-policies-raw-to-interim.py (hunk 5 above)
Comment (dpa9694) [reply code/data/multi_country/convert-policies-raw-to-interim.py:81]: good points!

Comment (dpa9694) [reply code/data/multi_country/convert-policies-raw-to-interim.py:81]: I tried to handle these issues in `clean_intensities_usa(df)`

File: code/data/multi_country/convert-policies-raw-to-interim.py (hunk 2 above)
Comment (bolliger32) [reply code/data/multi_country/convert-policies-raw-to-interim.py:9]: yup - a function. Cool! agreed.

**Pull request merged**